### ISSUE TITLE
Not available in Marketplace?

### ISSUE THREAD

**Issue by rulatir** [pulumi/pulumi-vscode-tools#23]
### What happened?

I tried to follow the README and install the plugin from marketplace. It is not available there.

### Example

![Image](https://github.com/user-attachments/assets/00968b90-4ee3-4941-8cad-66b98c2652c5)

### Output of `pulumi about`

Irrelevant, since this isn't about pulumi itself, it's about the VS Code extension's availability for installing.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mjeffryes** [comment]
I'm sorry you're having trouble finding it @rulatir; Are you sure there's nothing blocking your connection to the extensions marketplace? I can find the plugin with no problem with a new VScode install. 

Eg. using a github codespace instance:
<img width="410" alt="Image" src="https://github.com/user-attachments/assets/110ca66f-5b73-4a1b-9b55-d57321745295" />

**Comment by rulatir** [comment]
I don't believe anything is blocking my connection. When I type something common like `cmake` or `php` in the search field, I get expected results. `pulumi` - no results.

**Comment by mjeffryes** [comment]
I'm sorry, I don't have enough information to reproduce the problem you're experiencing. The extension is available in the VSCode extension marketplace by every means I can think to check. eg: https://marketplace.visualstudio.com/search?term=pulumi&target=VSCode&category=All%20categories&sortBy=Relevance

**Comment by rulatir** [comment]
I don't see an "install button" there.

Anyways, my vscode is an "open source build" (this is what the `code` package from the Arch official distro installs). Could that be the issue?

**Comment by mjeffryes** [comment]
Yeah, I suspect this comes down to some limitation of the "open source build" of VSCode or the architecture. You might have more luck diagnosing on a VSCode support forum? Unfortunately, I don't think this is something we'll be able to investigate further on our end.

**Comment by EronWright** [comment]
Thank-you for the report @rulatir.  According to the [ArchLinux wiki](https://wiki.archlinux.org/title/Visual_Studio_Code), there are indeed a number of variants. If you're using Code OSS, I believe it relies on the [Open VSX Registry](https://open-vsx.org/) as its extension marketplace. The Pulumi extensions aren't published there, and frankly I don't expect we'll get to it soon, so I'll close this issue for now.

I should mention that ArchLinux does seem to have an option to install "VS Code" which would assumedly fix the issue.

**Comment by smrqdt** [comment]
@EronWright Please reconsider this, Microsoft recently locked down the Marketplace even more by removing the download link. open-vsx is run by the Eclipse Foundation and used by a range of Code OSS-based or -compatible editors, like Arch Linux' Code package, VS Codium (both are mostly identical to VS Code with some privacy enhancements and without non-OSS-components), Eclipse Theia, Cursor and probably others. 

I’m also using the `code` package from the Arch Repository, switching to VS Code would be a pain. Microsoft only provides RPM and Debian packages, so any installation on Arch is a doctored version of these packages converted to an AUR build or flatpak. I tried that in the past and the user experience was bad. The other option is to patch Code OSS to use the marketplace, but that is also quite hacky (and against Microsoft’s ToS).

It [does not seem overly complicated to publish to open-vsx](https://github.com/EclipseFdn/open-vsx.org/wiki/Publishing-Extensions), and the Eclipse Foundation [even offers to auto-republish extensions](https://github.com/EclipseFdn/open-vsx.org/wiki/Auto-Publishing-Extensions) from the Marketplace if the author requests it.

Microsoft decided to deny competition from using their marketplace after establishing the ecosystem. Please also support the open and more privacy-friendly alternatives.

**Comment by auirarrazaval** [comment]
@EronWright @mjeffryes I'm joining @smrqdt in their request to reconsider this. While VSCode has dominated for years, Cursor and other Code based editors have gained significant popularity since you have closed this issue, and as stated above Microsoft has been bullish about their approach on how to mitigate people from leaving VSCode in favor of them.
